demo-app/src/main/java/io/opentelemetry/android/demo/ui/shop/AstronomyShopActivity.kt

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.android.demo.ui.shop
+
+import android.app.Activity
+import android.content.Intent
+import android.os.Bundle
+import androidx.activity.compose.setContent
+import androidx.appcompat.app.AppCompatActivity
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Surface
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.navigation.compose.NavHost
+import androidx.navigation.compose.composable
+import io.opentelemetry.android.demo.MainActivity
+import io.opentelemetry.android.demo.clients.ProductCatalogClient
+import io.opentelemetry.android.demo.theme.DemoAppTheme
+import io.opentelemetry.android.demo.ui.shop.cart.CartScreen
+import io.opentelemetry.android.demo.ui.shop.products.ProductDetails
+import io.opentelemetry.android.demo.ui.shop.products.ProductList
+
+class AstronomyShopActivity : AppCompatActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContent {
+            AstronomyShopScreen()
+        }
+    }
+}
+
+@Composable
+fun AstronomyShopScreen() {
+    val productsClient = ProductCatalogClient(LocalContext.current)
+    val products by remember { mutableStateOf(productsClient.get()) }
+    val context = LocalContext.current
+    val astronomyShopNavController = rememberAstronomyShopNavController()
+
+    DemoAppTheme {
+        Surface(
+            modifier = Modifier.fillMaxSize(),
+            color = MaterialTheme.colorScheme.background
+        ) {
+            Scaffold(
+                bottomBar = {
+                    BottomNavigationBar(
+                        items = listOf(BottomNavItem.Exit, BottomNavItem.List, BottomNavItem.Cart),
+                        currentRoute = astronomyShopNavController.currentRoute,
+                        onItemClicked = { route ->
+                            astronomyShopNavController.navController.navigate(route) {
+                                popUpTo(astronomyShopNavController.navController.graph.startDestinationId)
+                                launchSingleTop = true
```

Comment:
Any idea what this `launchSingleTop` does? It's new for me, so I'm just curious.

Reply:
Cool, makes sense thanks.